src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -897,12 +896,13 @@ static bool updateAttribute
 * original value of the attribute was different than the one used in the update (this is
 * important for ONCHANGE notifications). Otherwise it is false
 *
-* (Previous versions of this function used the return value for that, but we have
-* modified to make in similar to updateAttribute()
+* In addition, return value is as follows:
+* - true: there was an actual append change
+* - false: there was an append-as-update change
 *
```

Comment:
We extensively use the true/false return pattern in many functions in the mongoBackend. Typically, you only need to distinguish between two states, so althoug your solution is probably smarter, I think that it involves extra work (not only in this function, but the others using the similar pattern) with little gain.

Thus, I think this should be NTC